builtin/providers/aws/resource_aws_api_gateway_authorizer.go

```diff
@@ -0,0 +1,189 @@
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"identity_source": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"rest_api_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"credentials": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"result_in_ttl": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+
+			"identity_validation_expression": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayAuthorizerCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+	// Create the gateway
+	log.Printf("[DEBUG] Creating API Gateway Authorizer")
+
+	var credentials *string
+	if v, ok := d.GetOk("credentials"); ok {
+		credentials = aws.String(v.(string))
+	}
+
+	var result_in_ttl *int64
+	if v, ok := d.GetOk("result_in_ttl"); ok {
+		result_in_ttl = aws.Int64(int64(v.(int)))
+	}
+
+	var identity_validation_expression *string
+	if v, ok := d.GetOk("identity_validation_expression"); ok {
+		identity_validation_expression = aws.String(v.(string))
+	}
+
+	var err error
+	authorizer, err := conn.CreateAuthorizer(&apigateway.CreateAuthorizerInput{
+		AuthorizerUri:  aws.String(d.Get("authorizer_uri").(string)),
+		IdentitySource: aws.String(d.Get("identity_source").(string)),
+		Name:           aws.String(d.Get("name").(string)),
+		RestApiId:      aws.String(d.Get("rest_api_id").(string)),
+		Type:           aws.String(d.Get("type").(string)),
+
+		AuthorizerCredentials:        credentials,
+		AuthorizerResultTtlInSeconds: result_in_ttl,
+		IdentityValidationExpression: identity_validation_expression,
+	})
+	if err != nil {
+		return fmt.Errorf("Error creating API Gateway Authorizer: %s", err)
+	}
+
+	d.SetId(*authorizer.Id)
+	log.Printf("[DEBUG] API Gateway Authorizer ID: %s", d.Id())
+
+	return nil
+}
+
+func resourceAwsApiGatewayAuthorizerRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	log.Printf("[DEBUG] Reading API Gateway Deployment %s", d.Id())
+	out, err := conn.GetAuthorizer(&apigateway.GetAuthorizerInput{
+		RestApiId:    aws.String(d.Get("rest_api_id").(string)),
+		AuthorizerId: aws.String(d.Id()),
+	})
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "NotFoundException" {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	log.Printf("[DEBUG] Received API Gateway Authorizer: %s", out)
+	d.SetId(*out.Id)
+
+	return nil
+}
+
+func resourceAwsApiGatewayAuthorizerUpdateOperations(d *schema.ResourceData) []*apigateway.PatchOperation {
+	operations := make([]*apigateway.PatchOperation, 0)
+
+	if d.HasChange("description") {
```

Comment:
This looks like a copy-paste error - there isn't any `description` field in the schema and there's many fields that would need to be reflected here (i.e. those that are _updateable_).